src/platform/packages/shared/shared-ux/error_boundary/src/ui/error_boundary.tsx

```diff
@@ -39,6 +40,7 @@ class ErrorBoundaryInternal extends React.Component<
   }
 
   componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
+    apm.captureError(error);
     console.error('Error caught by Kibana React Error Boundary'); // eslint-disable-line no-console
     console.error(error); // eslint-disable-line no-console
```

Comment:
Q: should we merge https://github.com/elastic/kibana/pull/206565 and use core's logger instead?

UPDATE: nvm, that would require Core's deps to be provided whenever this component is used. LGTM

Reply:
one tiny nit: should we add attributes to the error to highlight that this is an error caught by this boundary?
Something similar to https://github.com/elastic/kibana/pull/206565/files#diff-5cd7840979b896fd0fef4674498f7ea8960decc3330f16d8610ed85fb5684c33R28.

WDYT?

Reply:
> WDYT?

Since this is a shared component, I worry that all errors will be attributed the same way and be hard for teams to distinguish their errors vs other teams' errors. I don't know how to verify on my own that the errors will be discoverable (teams can find their own errors) and actionable (teams can understand where the error originated from). Attributing the error as having been caught by the error boundary, as @afharo suggests, sounds like a good move in the right direction.

@dgieselaar is there anything you can add here? We should move the discussion from [the issue](https://github.com/elastic/observability-dev/issues/4222#issuecomment-2631615705) into this PR.

Reply:
from what I can tell, [any custom properties on the error will be added to `custom`](https://github.com/elastic/apm-agent-rum-js/blob/65ebcabe9cd1472d6b0b8da7b0a5489fe7d126f3/packages/rum-core/src/error-logging/error-logging.js#L98) - @vigneshshanmugam can you confirm? If that holds, you could mutate the error object and set a property that indicates it's been caught by the error boundary, or you could add something to the stack.

Reply:
That's correct. It would be under 'context.custom.foo' as part of the document.